packages/svelte/src/internal/client/runtime.js

```diff
@@ -722,11 +727,14 @@ function process_effects(effect, collected_effects) {
  */
 export function yield_updates(fn) {
 	const previous_scheduler_mode = current_scheduler_mode;
-	// If we're calling yield_updates, and we've already yielded some updates – then it's likely
-	// that the event might might try and read from the UI. In order for this to be glitch free,
-	// we can flush any changes, forcing the UI to be up-to-date, so any reads from the UI work as
-	// expected.
-	if (previous_scheduler_mode !== FLUSH_YIELD && is_yield_task_queued && !is_micro_task_queued) {
+	// If we're calling yield_updates and there is already an active yield in progress (is_yield_task_active)
+	// then it's likely that the event might might try and read from the UI. Additionally, we might be dealing with
+	// event that was sync called from another event flow, thus isTrusted() will be true – such as a submit event from
+	// a subsequent click event. In either case, the UI needs to be up-to-date, so we flush any pending changes.
```

Comment:
Can you elaborate a bit why isTrusted is a good hint for "event was sync called from another event flow"? From my understanding isTrusted only means "the user did this". How can we infer the sync-called state from it?

Reply:
Well we can't fully, we can only infer that given that `is_yield_task_queued` is true, that some event before it has run and now if we have another event that is trusted it might have been called from that. Whilst the `click` won't be trusted, as it was done via `element.click()` the `submit` will be trusted.